e2e/kube.go

```diff
@@ -637,3 +632,52 @@ func nvidiaDevicePluginDaemonSet() *appsv1.DaemonSet {
 		},
 	}
 }
+
+func podEnableAMDGPUResource(s *Scenario) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-amdgpu-device-plugin", s.Runtime.KubeNodeName),
+			Namespace: defaultNamespace,
+		},
+		Spec: corev1.PodSpec{
+			PriorityClassName: "system-node-critical",
+			NodeSelector: map[string]string{
+				"kubernetes.io/hostname": s.Runtime.KubeNodeName,
+			},
+			Containers: []corev1.Container{
+				{
+					Name:  "amdgpu-device-plugin-container",
+					Image: "rocm/k8s-device-plugin",
```

Comment:
should pull from MCR, dockerhub pull isn't stable. (assuming this is from docker.io)

Reply:
I'll need to add it to MCR at some stage. I don't think it's available there yet.

Reply:
there are SFI to looks for that, we lost the contract preventing us from getting throttled on docker.io so expect throttling. 

We have an ACR for E2E, we should setup a remote/cache for this image and use our own ACR